### DISCUSSION TITLE
Changing colors on runtime?

### DISCUSSION THREAD

**Discussion by NathanTe** [saadeghi/daisyui#777] [Q&A] [answered]
Hey guys,

quick question is there a way to dynamically change colors on runtime? I would like to be able to call my api get color values and change the colors of daisyui?
I know it is possible to define different themes but I don't know the colors in advance so this is not an option.

any help would be appreciated . 
@saadeghi

**Comment by saadeghi** [comment, marked as answer]
Each color is a CSS variable with HSL values.  
You can see the colors of all built-in themes here: https://unpkg.com/browse/daisyui@2.14.3/dist/themes.css  
And [here](https://daisyui.com/docs/colors/) you can see the list of all color names and their variable name.  

And you can change the color on runtime by changing the value of CSS variable.  
If your color has a different format like hex, RGB, etc... you need a JS function (or [a node package](https://www.npmjs.com/package/color)) to convert the format and extract the HSL values.

**Reply by NathanTe** [reply]
So I would be able to override the --p variable and te primary color would change everywhere?

**Reply by saadeghi** [reply]
Yes.  
https://play.tailwindcss.com/uTdCPT4NWj?file=css

**Reply by NathanTe** [reply]
Thx for the example got it working :)

**Comment by Torbet** [comment]
@saadeghi I see you now use `oklch` instead of HSL, I'm struggling to find a good library to convert between hex and lch, any suggestions?

**Reply by saadeghi** [reply]
This is a really good package:  
https://culorijs.org/

By the way, `lch` is different from `oklch`. daisyUI 4 uses `oklch`

**Reply by jamesladd** [reply]
Does this still work in V5.x ?

**Reply by pdanpdan** [reply]
Yes. You find the [color variables you want to change](https://daisyui.com/docs/colors/#list-of-all-daisyui-color-names) and set them on the desired element (:root).
Make sure to also set `color-scheme: light` or `color-scheme: dark`.
